opentelemetry-instrumentation-tower/src/lib.rs

```diff
@@ -394,29 +437,65 @@ where
         let url_scheme_kv = KeyValue::new(URL_SCHEME_LABEL, scheme);
 
         let method = req.method().as_str().to_owned();
-        let method_kv = KeyValue::new(HTTP_REQUEST_METHOD_LABEL, method);
+        let method_kv = KeyValue::new(HTTP_REQUEST_METHOD_LABEL, method.clone());
 
         #[allow(unused_mut)]
         let mut route_kv_opt = None;
         #[cfg(feature = "axum")]
         if let Some(matched_path) = req.extensions().get::<MatchedPath>() {
-            route_kv_opt = Some(KeyValue::new(
-                HTTP_ROUTE_LABEL,
-                matched_path.as_str().to_owned(),
-            ));
+            let route = matched_path.as_str().to_owned();
+            route_kv_opt = Some(KeyValue::new(HTTP_ROUTE_LABEL, route.clone()));
         };
 
         // Extract custom request attributes
         let custom_request_attributes = self.request_extractor.extract_attributes(&req);
 
+        // Start tracing span
+        let mut span_attributes = vec![
+            KeyValue::new(semconv::trace::HTTP_REQUEST_METHOD, method.clone()),
+            url_scheme_kv.clone(),
+            KeyValue::new(semconv::attribute::URL_PATH, req.uri().path().to_string()),
+            KeyValue::new(semconv::trace::URL_FULL, req.uri().to_string()),
+        ];
+
+        if let Some(user_agent) = req
+            .headers()
+            .get("user-agent")
+            .and_then(|v| v.to_str().ok())
+        {
+            span_attributes.push(KeyValue::new(
+                semconv::trace::USER_AGENT_ORIGINAL,
+                user_agent.to_string(),
+            ));
+        }
+
+        span_attributes.extend(custom_request_attributes.clone());
+
+        let span_name = format!("{} {}", method, req.uri().path());
+
+        let tracer = match &self.state.tracer_provider {
+            Some(tp) => {
+                BoxedTracer::new(Box::new(tp.tracer("opentelemetry-instrumentation-tower")))
+            }
+            None => global::tracer("opentelemetry-instrumentation-tower"),
+        };
+
+        let span = tracer
+            .span_builder(span_name)
+            .with_kind(SpanKind::Server)
+            .with_attributes(span_attributes)
+            .start(&tracer);
+
+        let cx = OtelContext::current_with_span(span);
```

Comment:
do we attach/activate the context? or just store it ourselves? Normally, this should be just activated, so any subsequent code can access it.

Reply:
https://github.com/open-telemetry/opentelemetry-rust-contrib/pull/431#discussion_r2561682483